spec/fn_util_spec.lua

```diff
@@ -0,0 +1,61 @@
+--- Triple Comment to Enable our LLS Plugin
+describe('function utils', function()
+	local FnUtil = require('Module:FnUtil')
+
+	describe('memoize', function()
+		it('check', function()
+			local calledCount = 0
+			local square = FnUtil.memoize(function(x)
+				calledCount = calledCount + 1
+				return x ~= nil and x * x or nil
+			end)
+
+			assert.are_equal(0, calledCount)
+			assert.are_equal(4, square(2))
+			assert.are_equal(1, calledCount)
+			assert.are_equal(4, square(2))
+			assert.are_equal(1, calledCount)
+			assert.are_equal(9, square(3))
+			assert.are_equal(2, calledCount)
+			assert.are_equal(9, square(3))
+			assert.are_equal(2, calledCount)
+			assert.is_nil(square(nil))
```

Comment:
intended to have the same line 3 times here?

Reply:
wouldn't 2 be enough?
anyways lgtm

Reply:
Just took the old test 🤷